File: Delgado-Alarcon-Alan_Ignacio/procesos/C/processB.c
#include <stdio.h>
#include <unistd.h>
#include <time.h>
#include <stdlib.h>

void hijo(){
    sleep(2);
    printf("quien?... te pregunto");
}

void padre(){
    sleep(10);
    system("ps");
    printf("hijos de...");
}

int main(){
    for(int i=1; i<=3; i++){
        pid_t pid = fork();
        switch(pid) {
            case 1:
                printf("ERROR\n");
                return 1;
            case 0:
                hijo();
                exit(1);
            default:
                padre();
        }
    }
}

File: Delgado-Alarcon-Alan_Ignacio/procesos/Python/processB.py
import os
import time

def hijo():
    time.sleep(2)
    print("quien?... te pregunto")

def padre():
    time.sleep(10)
    os.system("ps")
    print("hijos de...")

for i in range(1, 4):
    try:
        pid = os.fork()
        if pid == 0: 
            hijo()
            os._exit(0)
        else: 
            padre()
    except OSError as e:
        print("ERROR")
        exit(1)
